FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+import { Link } from 'react-router-dom';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { useState, useMemo } from 'react';
+import { useSWRConfig } from 'swr';
+import useSWRMutation from 'swr/mutation';
+import styles from './CompanyCard.module.css';
+import { useUser } from '../../hooks';
+import PropTypes from 'prop-types';
+import { Tooltip, Badge } from 'antd';
+
+export default function CompanyCard({ data, isAuthorized }) {
+  CompanyCard.propTypes = {
+    data: PropTypes.object,
+    isAuthorized: PropTypes.bool,
+  };
+  const [isSaved, setIsSaved] = useState(data && data.is_saved);
+  const { mutate } = useSWRConfig();
+  const profile = useMemo(() => {
+    return {
+      id: data?.id,
+      personId: data?.person,
+      name: data?.name,
+      region: data?.region,
+      categories: data?.categories,
+      activities: data?.activities,
+      logo: data?.logo_image,
+      banner: data?.banner_image,
+      founded: data?.founded,
+      isSaved: data?.is_saved,
+    };
+  }, [data]);
+
+  const { user } = useUser();
+  const ownProfile = user && user.id === profile.personId;
+
+  async function sendRequest(url, { arg: data }) {
+    const authToken = localStorage.getItem('Token');
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Token ${authToken}`,
+      },
+      body: JSON.stringify(data),
+    }).then();
+  }
+
+  const { trigger } = useSWRMutation(
+    `${process.env.REACT_APP_BASE_API_URL}/api/saved-list/`,
+    sendRequest
+  );
+
+  const handleClick = async () => {
+    try {
+      await trigger(
+        { company_pk: profile.id },
+        { optimisticData: () => setIsSaved(!isSaved) }
+      );
+      mutate(
+        (key) => typeof key === 'string' && key.startsWith('/api/profiles/'),
+        {
+          revalidate: true,
+        }
+      );
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const filledStar = (
+    <StarFilled
+      className={styles['star']}
+      onClick={handleClick}
+      data-testid="star"
+    />
+  );
+  const outlinedStar = (
+    <StarOutlined
+      className={styles['star']}
+      onClick={handleClick}
+      data-testid="emptystar"
+    />
+  );
+
+  const CategoryBadges = ({ categories }) => {
+    return (
+      <div>
+        {categories
+          ? categories.map((category) => (
+              <Badge
+                title=""
+                key={category.id}
+                size="medium"
+                count={category.name.toUpperCase()}
+                style={{
+                  backgroundColor: '#1F9A7C',
+                  fontWeight: 600,
+                  fontFamily: 'Inter',
+                  fontSize: 10,
+                  margin: 5,
+                }}
```

Comment:
Not a good practice to store inline styles. Could we achieve it with css? If not, we can at least move styles to constant or incapsulate it as a new component if its going to be reusable

Reply:
Thank your for feedback. Done.